### PR TITLE
[GPU] Resolve assertion of vsr int8 batch32

### PR DESCRIPTION
**Pull request by byungilm**
+ Bugfix
+ Unexpected removal of reorder in front of narrow output conv

Signed-off-by: Min, Byungil <byungil.min@intel.com>

### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/remove_redundant_reorders.cpp [hunk 1]
```diff
@@ -90,6 +90,22 @@ void remove_redundant_reorders::run(program& p) {
 
                 if (usr->is_type<fully_connected>())
                     recalc_list.push_back(usr);
+
+                // Not to fuse reorder if this removal changes input format of its next node which has reuse in fused_op
+                if (all_users_fuse == true && usr->get_preferred_impl_type() == impl_types::onednn) {
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/remove_redundant_reorders.cpp:95]: all_users_fuse is always true here.

Comment (isanghao) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/remove_redundant_reorders.cpp:95]: BTW, I think this needs to be placed within `can_fuse_reorder` function.

**Review by isanghao — CHANGES_REQUESTED** [review]

File: inference-engine/thirdparty/clDNN/api/cldnn/primitives/eltwise.hpp [hunk 2]
```diff
@@ -168,6 +168,15 @@ struct eltwise : public primitive_base<eltwise> {
         return false;
     }
 
+    bool can_reuse_eltwise_mem(layout input_layout, layout output_layout) const {
```
[the remark this answers is not in the document]
Comment (vladimir-paramuzov) [reply inference-engine/thirdparty/clDNN/api/cldnn/primitives/eltwise.hpp:171]: I think program_helpers is the right place for this method

File: inference-engine/thirdparty/clDNN/src/program_helpers.cpp [hunk 3]
```diff
@@ -175,4 +175,15 @@ std::pair<bool, bool> program_helpers::are_layouts_identical(layout const& l1, l
 
     return {false, false};
 }
+
+// check if input and output layouts are indentical to reuse memory in fused_ops of onednn
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/program_helpers.cpp:179]: nit: `indentical` -> `identical`

**Review by isanghao — APPROVED** [review]

File: inference-engine/thirdparty/clDNN/src/program_helpers.cpp (hunk 3 above)
Comment (byungilm) [reply inference-engine/thirdparty/clDNN/src/program_helpers.cpp]: Fixed

**Review by sshlyapn — APPROVED** [review]

**Pull request merged**